url_finder/src/provider_endpoints.rs

```diff
@@ -1,19 +1,86 @@
+use alloy::sol_types::SolType;
+use alloy::{
+    network::TransactionBuilder,
+    primitives::{Address, Bytes},
+    providers::{Provider, ProviderBuilder},
+    rpc::types::eth::TransactionRequest,
+    sol,
+    sol_types::SolCall,
+};
+
+use color_eyre::{Result, eyre::eyre};
 use tracing::{debug, error};
 
 use crate::{
+    ErrorCode, ResultCode,
     cid_contact::{self, CidContactError},
-    lotus_rpc, multiaddr_parser, ErrorCode, ResultCode,
+    lotus_rpc, multiaddr_parser,
 };
 
+sol! {
+    struct PeerData {
+        string peerID;
+        bytes multiaddrs;
+    }
+
+    function getPeerData(uint64 minerID) view returns (PeerData);
+}
+
+pub async fn valid_curio_provider(address: &str) -> Result<Option<String>> {
+    let rpc_url = "https://api.node.glif.io/rpc/v1";
+
+    let rpc_provider = ProviderBuilder::new()
+        .connect(rpc_url)
+        .await
+        .map_err(|err| eyre!("Building provider failed: {}", err))?;
```

Comment:
Is this `map_err` needed here? Doesnt seem to add anything valueble and err already implements std error?